internal/integration/integration_test.go

```diff
@@ -53,316 +69,323 @@ func TestIntegration(t *testing.T) {
 		OnlyLocalProvenance: false,
 	}
 
-	keys := util.GenerateExposureKeys(3, -1, false)
+	for _, tc := range cases {
 
-	// Publish 3 keys
-	payload := &verifyapi.Publish{
-		Keys:              keys,
-		HealthAuthorityID: "com.example.app",
-	}
-	jwtCfg.ExposureKeys = keys
-	verification, salt := testutil.IssueJWT(t, jwtCfg)
-	payload.VerificationPayload = verification
-	payload.HMACKey = salt
-	var revisionToken string
-	if resp, err := client.PublishKeys(payload); err != nil {
-		t.Fatal(err)
-	} else {
-		if resp.RevisionToken == "" {
-			t.Fatal("empty revision token")
-		}
-		revisionToken = resp.RevisionToken
-		t.Logf("response: %+v", resp)
-	}
+		keys := util.GenerateExposureKeys(3, -1, false)
 
-	// Assert there are 3 exposures in the database
-	{
-		exposures, err := getExposures(db, criteria)
-		if err != nil {
-			t.Fatal(err)
+		// Publish 3 keys
+		payload := &verifyapi.Publish{
+			Keys:              keys,
+			HealthAuthorityID: "com.example.app",
 		}
-		if got, want := len(exposures), 3; got != want {
-			t.Fatalf("expected %v to be %v", got, want)
+		jwtCfg.ExposureKeys = keys
+		jwtCfg.JWTWarp = tc.JWTWrap
+		verification, salt := testutil.IssueJWT(t, jwtCfg)
+		payload.VerificationPayload = verification
+		payload.HMACKey = salt
+		var revisionToken string
+		if resp, err := client.PublishKeys(payload); err != nil {
+			if !strings.Contains(err.Error(), tc.Error) {
```

Comment:
I would prefer an error comparison instead of strings.Contains

Reply:
Yes, please use `errors.Is` if possible.

Reply:
For some reason the error comparison always fails

This is what I get
```
expected error: failed to POST /publish: response was not 200 OK: {"error":"unable to validate diagnosis verification: Token used before issued","code":"health_authority_verification_certificate_invalid"}: , got: failed to POST /publish: response was not 200 OK: {"error":"unable to validate diagnosis verification: Token used before issued","code":"health_authority_verification_certificate_invalid"}: 
```

I am guessing it's because the way the error is generated by the client.go file.
I am not sure what exactly is happening here

Reply:
`client.go` uses `%w` wrapping error here https://github.com/google/exposure-notifications-server/blob/c5f5cf67733a61676de4d046a39df40ee2c91dd1/internal/integration/client.go#L47, can you try using `errors.Is` with wanted error below?
```
wantErr := errors.New("unable to validate diagnosis verification: Token used before issued")
```

Reply:
I have tried that too.
Tried again. Still the same result.

```
wantErr := fmt.Errorf(`response was not 200 OK: {"error":"unable to validate diagnosis verification: Token used before issued","code":"health_authority_verification_certificate_invalid"}`)
				if !errors.Is(err, wantErr) {
					t.Fatalf("expected error: %s, got: %s,", err, wantErr)
				}
```

Reply:
https://stackoverflow.com/questions/62441960/golang-error-wrap-unwrap-type-checking-with-errors-is

Looks like that errors.Is does an actual instance match and this will definitely not work in this case.

Shall we stay with the error message check or are there other ways to do this?

Reply:
the other option could be having an exported var in client.go that stores the error and we use that to compare.
But I don't like this option